### PR TITLE
thirdparty: update `zip` to commit 6f2116d from https://github.com/kuba--/zip/

### PR DESCRIPTION
**Pull request by ttytm**

### CODE REVIEW

File: thirdparty/zip/miniz.h [hunk 1]
```diff
@@ -4999,15 +5015,8 @@ static int mz_mkdir(const char *pDirname) {
...
 
 #ifndef MINIZ_NO_TIME
-
-#if (defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)) && !defined(FREEBSD)
-#include <utime.h>
-#else
 #include <sys/utime.h>
 #endif
-
-#endif
```
Comment (spytheman) [inline thirdparty/zip/miniz.h:5009]: That was needed on FreeBSD. Does it work without the patch now?

File: thirdparty/zip/miniz.h [hunk 2]
```diff
@@ -5041,15 +5048,8 @@ static int mz_mkdir(const char *pDirname) {
...
 
 #elif defined(__TINYC__)
-
 #ifndef MINIZ_NO_TIME
-
-#if (defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)) && !defined(FREEBSD)
-#include <utime.h>
-#else
 #include <sys/utime.h>
-#endif
```
Comment (spytheman) [inline thirdparty/zip/miniz.h:5051]: Same, are you sure that patch is not needed anymore?

Comment (spytheman) [reply thirdparty/zip/miniz.h:5051]: I just checked on a FreeBSD vps:
```
1[08:58:52]ec2-user@freebsd: (compress/update-zip) ~/v $ v cmd/tools/vcompress.v
builder error: 'sys/utime.h' not found
1[08:58:58]ec2-user@freebsd: (compress/update-zip) ~/v $ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
0[08:59:01]ec2-user@freebsd: (master) ~/v $ v cmd/tools/vcompress.v
0[08:59:03]ec2-user@freebsd: (master) ~/v $
```

-> it is still needed for tcc on FreeBSD.

**Comment by spytheman** [comment]
Where is it sourced from?
I looked at <https://github.com/richgel999/miniz/releases>, but the version there in the .zip file, contains both a miniz.h and miniz.c files, and a lot more differences 🤔 .

**Comment by spytheman** [comment]
A specific commit can work too. I just want to know what it is, so that I can include it in the PR's merge commit message, so that in the future, it would be easier to update, and see what the diffs are.

**Comment by ttytm** [comment]
`6f2116d`
https://github.com/kuba--/zip/tree/6f2116d77cdcbe544e4dbbf1e53895596ed14b89

**Comment by spytheman** [comment]
Thanks.

**Pull request merged**